### PR TITLE
Refactor docs

### PR DESCRIPTION
**Pull request by akarve**
* Consolidate Python and CLI docs to one page
* Remove dedicated CLI page
* Mention teams in API docs (and update dedicated teams section)

### CODE REVIEW

File: docs/teams.md [hunk 1]
```diff
@@ -1,72 +1,69 @@
...
 Send a user a reset-password email.
 ```sh
 quilt user reset-password TEAM USERNAME
 ```
 
-## `quilt audit`
+### `quilt audit`
 Audit events relating to a user or package.
 ```sh
 quilt audit USER_OR_PACKAGE
```
Comment (asah) [inline docs/teams.md]: so package names and user names share the same namespace, i.e. one reserves the other?  seems weird/bad...

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+### Example
+```
+packages:
+  - vgauthier/DynamicPopEstimate   # get latest
+  - danWebster/sgRNAs:a972d92      # get a specific version via hash
+  - akarve/sales:tag:latest        # get a specific version via tag
+  - asah/snli:v:1.0                # get a specific version via version
+```
+
+# API
```
Comment (asah) [inline docs/api.md]: API?  isn't this CLI?

File: docs/teams.md (hunk 1 above)
Comment (akarve) [reply docs/teams.md]: package names (handles really) always start with usernames. and only package handles can contain a slash. so there's no ambiguity.

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+Commands that take hashes support "short hashes", up to uniqueness.
+```sh
+quilt install akarve/examples -x 4594b5
+# matches hash 4594b58d64dd9c98b79b628370618031c66e80cbbd1db48662be0b7cac36a74e
+```
+ In practice, 6-8 characters is sufficient to achieve uniqueness.
+
+## Requirements file (quilt.yml)
+```sh
+$ quilt install [@filename]
```
Comment (meffij) [inline docs/api.md]: Remember to use CAPS for metasyntax, i.e. `[@FILENAME]`

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+# quilt.yml is the default if @filename is absent
+```
+
+Installs a list of packages specified by a YAML file. The YAML file must contain a `packages` node with a list of packages: 
+```yaml
+packages:
+  - USER/PACKAGE[/SUBPACKAGE][:hash|:tag|:version][:HASH|TAG|VERSION]
```
Comment (meffij) [inline docs/api.md]: [:h[ash]|:t[ag]|:v[ersion]] since we allow abbreviations

File: docs/api.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+| `quilt rm USER/PACKAGE` | `quilt.rm("USER/PACKAGE")` | Remove a package from local storage (but not from the registry) |
+
+## Registry search
+| Command line | Python | Description |
+| --- | --- | --- |
+| `quilt search "SEARCH STRING"` |  `quilt.search("SEARCH STRING")` | Search registry for packages by user or package name |
+
+## Import and use data
+```python
+from quilt.data.USER import PACKAGE
```
Comment (meffij) [inline docs/api.md]: Needs to include information about importing team packages

**Pull request merged**